src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -391,6 +400,115 @@ public void createMissingClass(UnsolvedClass missedClass) {
     }
   }
 
+  /**
+   * This method capitalizes a string. For example, "hello" will become "Hello".
+   *
+   * @param string the string to be capitalized
+   * @return the capitalized version of the string
+   */
+  public static String toCapital(String string) {
+    return string.substring(0, 1).toUpperCase() + string.substring(1);
+  }
+
+  /**
+   * This method checks if a string is capitalized
+   *
+   * @param string the string to be checked
+   * @return true if the string is capitalized
+   */
+  public static boolean isCapital(String string) {
+    String capitalizedString = toCapital(string);
+    return string.equals(capitalizedString);
+  }
+
+  /**
+   * This method checks if a string is a class path.
+   *
+   * @param potentialClassPath the string to be checked
+   * @return true if the string is a class path
+   */
+  public static boolean isAClassPath(String potentialClassPath) {
+    String[] elements = potentialClassPath.split(".");
+    int elementsCount = elements.length;
+    return elementsCount > 1 && isCapital(elements[elementsCount - 1]);
+  }
+
+  /**
+   * Given a class path, this method will create a simple instance of UnsolvedClass for that class
+   * path
+   *
+   * @param classPath the class path
+   * @return the corresponding instance of UnsolvedClass
+   */
+  public static UnsolvedClass getSimpleSyntheticClassFromClassPath(String classPath) {
+    if (!isAClassPath(classPath)) {
+      throw new RuntimeException(
+          "Check with isAClassPath first before using getSimpleSyntheticClassFromClassPath");
+    }
+    String[] elements = classPath.split(".");
+    // the last element of a class path is the simple name of that class
+    @SuppressWarnings("signature")
+    @ClassGetSimpleName String className = elements[elements.length - 1];
```

Comment:
Use a helper method (with a suppression) rather than suppressing this directly, so that the conversation from "class path" to simple name only needs to happen in one place.